index.php

```diff
@@ -8,13 +8,11 @@
 require_once(__DIR__ . '/OutputXMLHelper.php');
 $outputXmlHelper = new OutputXMLHelper();
 // Get URL params
-$start = $outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET);
-$count = $outputXmlHelper->getUrlParam($outputXmlHelper::COUNT_NAME, $outputXmlHelper::DEFAULT_COUNT_VALUE, $_GET);
+$start = (int)$outputXmlHelper->getUrlParam($outputXmlHelper::START_NAME, $outputXmlHelper::DEFAULT_START_VALUE, $_GET, FILTER_VALIDATE_INT);
```

Comment:
Casting to `int` shouldn't be necessary anymore, now that the type is checked.

Reply:
It still is needed, because the _GET param will always return a string. So I need to manually convert the string, containing the integer value, to an integer.

`count=2` → `$_GET['count'] = '2'`.

Reply:
`filter_var` does not only validate the type but also converts it if possible. So `getUrlParam` should run the value through `filter_var` not only for validation but also for converting the type.